Server/controllers/leave.controller.js

```diff
@@ -1,18 +1,92 @@
 "use strict";
+const leavesDataLayer = require('../datalayer/leave.datalayer');
 /**
  * Class representing leaves controller
  *
  * @class leavesController
  */
 class leavesController {
 
-    static getLeavesByUserId (req, res) {}
+    static async getLeavesByUserId (req, res) {
+        const userId = req.query.user;
+        if (!userId) {
+            return res.status(400);
+        }
+        let leaves;
+        try {
+            leaves = await leavesDataLayer.getLeavesByUserId(userId);
+        } catch (err) {
+            console.log(err.message);
+            return res.status(500);
+        }
+        return res.status(200).json(leaves);
+    }
 
-    static createLeave (req, res) {}
+    static async createLeave (req, res) {
+        let createdLeave;
+        let {endDate, reason, startDate, userId} = req.body;
+        if (!endDate || !reason || !startDate || !userId) {
+            return res.status(400);
+        }
+        const newLeave = {
+            "startDate": startDate,
+            "endDate": endDate,
+            "reason": reason,
+            "userId": userId
+        }
```

Comment:
In shorthand, we can write like 
`const newLeave = {startDate, endDate, reason, userId} `

Reply:
Changed.